modules/images/webp-uploads/load.php

```diff
@@ -683,3 +683,121 @@ function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Pos
+		// Add sources to original image metadata.
+		$extension                  = explode( '|', $allowed_mimes[ $targeted_mime ] );
+		$filename_without_extension = pathinfo( $file, PATHINFO_FILENAME );
+		$image_file                 = path_join( $original_directory, "$filename_without_extension.{$extension[0]}" );
+
+		if ( ! file_exists( $image_file ) ) {
+			continue;
+		}
+
+		$metadata['sources'][ $targeted_mime ] = array(
+			'file'     => wp_basename( $image_file ),
+			'filesize' => filesize( $image_file ),
+		);
+
+		foreach ( $metadata['sizes'] as $size => $size_details ) {
+			// Add sources to resized image metadata.
+			$filename_without_extension = pathinfo( $size_details['file'], PATHINFO_FILENAME );
+			$image_file                 = path_join( $original_directory, "$filename_without_extension.{$extension[0]}" );
+
+			if ( ! file_exists( $image_file ) ) {
+				continue;
+			}
+
+			$metadata['sizes'][ $size ]['sources'][ $targeted_mime ] = array(
+				'file'     => wp_basename( $image_file ),
+				'filesize' => filesize( $image_file ),
+			);
+		}
+	}
+
+	return $metadata;
+}
+
+/**
+ *
+ * Creates additional image formats when original image is edited.
+ *
+ * @since n.e.x.t
+ *
+ * @param bool|null       $override  Value to return instead of saving. Default null.
+ * @param string          $file      Name of the file to be saved.
+ * @param WP_Image_Editor $image     The image editor instance.
+ * @param string          $mime_type The mime type of the image.
+ * @param int             $post_id   Attachment post ID.
+ *
+ * @return bool|null
+ */
+function webp_uploads_update_image_onchange( $override, $file, $image, $mime_type, $post_id ) {
+	if ( null !== $override ) {
+		return $override;
+	}
+
+	if ( empty( webp_uploads_get_upload_image_mime_transforms()[ $mime_type ] ) ) {
+		return $override;
+	}
+
+	$valid_mime_transforms = webp_uploads_get_upload_image_mime_transforms()[ $mime_type ];
+	$original_directory    = pathinfo( $file, PATHINFO_DIRNAME );
+	$filename              = pathinfo( $file, PATHINFO_FILENAME );
+	$sizes                 = wp_get_registered_image_subsizes();
+	$allowed_mimes         = array_flip( wp_get_mime_types() );
+
+	foreach ( $valid_mime_transforms as $targeted_mime ) {
+		if ( $targeted_mime === $mime_type ) {
+			continue;
+		}
+
+		if ( ! isset( $allowed_mimes[ $targeted_mime ] ) || ! is_string( $allowed_mimes[ $targeted_mime ] ) ) {
+			return $override;
+		}
+
+		if ( ! wp_image_editor_supports( array( 'mime_type' => $targeted_mime ) ) ) {
+			return $override;
+		}
+
+		$extension   = explode( '|', $allowed_mimes[ $targeted_mime ] );
+		$destination = trailingslashit( $original_directory ) . "{$filename}.{$extension[0]}";
+
+		if ( is_wp_error( $image->save( $destination, $targeted_mime ) ) ) {
+			return $override;
+		}
+
+		$_sizes = array();
+
+		foreach ( $sizes as $size => $size_details ) {
+			$_sizes[ $size ] = $size_details;
+		}
+
+		$image->multi_resize( $_sizes );
```

Comment:
There is a problem with generating the sub-sizes here: In this code we generate new images for all sizes, however that may not be intended. We need to instead only generate the same sizes that were also generated for the original MIME by WordPress core.

To do that, I suggest we remove the process of generating additional sizes from _this_ function and instead handle it within the `wp_update_attachment_metadata` hook below. That would be better for the following reason:
* The function here is hooked into `wp_save_image_editor_file`. At this point, WordPress core has saved the edited (full) image, but it hasn't generated any sub-sizes for it yet, so generating the additional sub-sizes here is too early.
* By moving sub-sizes generation into the `wp_update_attachment_metadata` hook callback, we can compare the updated sizes (`$metadata['sizes']`) with the previous sizes (`wp_get_attachment_metadata( $post_meta_id )['sizes']`). Then, we should create the additional MIME type versions for _only_ the sizes that have changed.
* We can compare the file name for each size; if it hasn't changed in the new metadata compared to the other metadata, that sub-size wasn't generated, so we shouldn't do it either.

Side note: The `foreach` loop right now does nothing useful anyway, as it just copies over `$sizes` into `$_sizes`, without change.